libraries/app/api.cpp

```diff
@@ -328,6 +328,10 @@ namespace graphene { namespace app {
              start = node.operation_id;
        } catch(...) { return result; }
 
+       auto es = _app.get_plugin<elasticsearch::elasticsearch_plugin>("elasticsearch");
+       if(es)
```

Comment:
Need to check if `_elasticsearch_operation_string` is enabled too, because otherwise we can't return proper history entries from ES.

What we *actually* need to check is if the entries we're looking up have the `op` field, i. e. if `_elasticsearch_operation_string` has always been enabled on this database. We can't know that from current settings, therefore I think it would be better to use an explicit config flag that says if we want to use ES for queries or not.

That said, I think an important use-case is to run a node that queries ES but does not write into it (an ES cluster that is written by only one node can be used by many for queries). So perhaps you should move ES queries into a separate plugin.

Reply:
What about having a `mode` enum field with 3 options:
- `only_save` - Will connect to apply block, `elasticsearch-operation-string` can be on or off,  same as `elasticsearch-operation-object`.
- `only_query` - Will skip connection to apply_block altogether, saving options as `elasticsearch-operation-string` and `elasticsearch-operation-object` will be ignored.
- `both` - Will connect to apply_block, `elasticsearch-operation-string` will be forced to be on, `elasticsearch-operation-object` will remain optional.

A new plugin just for the queries is still a doable option however i am not sure if it will be the best to setup. I am adding @clockworkgr and @sschiessl-bcp to the discussion as they have public ES nodes.

Reply:
That would work as well, unless you want different ES server URLs for reading and writing.